### PR TITLE
Pet Ferrets

### PR DESCRIPTION
**Pull request by klorpa**
#### Summary
Content "Adds domesticated ferrets as encounterable animals"

#### Purpose of change
Adds small amount of content for players.

#### Testing
I still need to calculate the proper volumes.

#### Additional context
Please leave any suggestions, questions, or comments that you like.

### CODE REVIEW

**Comment by Maleclypse** [comment]
You're going to want to rebase on a current version of the master repo.  Bunch of unrelated changes in your PR.

File: data/json/monsters/mammal.json [hunk 1]
```diff
@@ -2954,5 +2954,66 @@
...
+    "harvest": "mammal_tiny",
+    "petfood": {
+      "food": [ "CATFOOD" ],
+      "feed": "The %s seems to like you!  It lets you pat its head and seems friendly.",
+      "pet": "The %s lets out a happy squeak as you give its head a small scritch."
+    },
+    "families": [ "prof_intro_biology" ],
+    "reproduction": { "baby_monster": "mon_ferret_kit", "baby_count": 1, "baby_timer": 100 },
+    "fear_triggers": [ "SOUND", "PLAYER_CLOSE" ],
+    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "SWIMS", "WARM", "CANPLAY", ]
```
Comment (TheShadowFerret) [inline data/json/monsters/mammal.json:2990]:
```suggestion
    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "SWIMS", "WARM", "CANPLAY" ]
```

**Review by TheShadowFerret — COMMENTED** [review]
~~Whomst dared to ping me by mentioning ferrets~~

File: data/json/monstergroups/wilderness.json [hunk 2]
```diff
@@ -976,6 +976,8 @@
       },
       { "monster": "mon_weasel", "weight": 5, "cost_multiplier": 5 },
       { "monster": "mon_weasel", "weight": 15, "cost_multiplier": 5, "conditions": [ "NIGHT" ] },
+      { "monster": "mon_ferret", "weight": 4, "cost_multiplier": 5 },
+      { "monster": "mon_ferret", "weight": 14, "cost_multiplier": 5, "conditions": [ "NIGHT" ] },
```
Comment (LyleSY) [inline data/json/monstergroups/wilderness.json:980]:
```suggestion
```
This use of conditions is buggy and isn't recommended for overmap spawns

Comment (LyleSY) [reply data/json/monstergroups/wilderness.json]: Fixed fixing the merge conflicts

**Review by LyleSY — APPROVED** [review]

**Review by LyleSY — APPROVED** [review]

**Pull request merged**